UIManager.js

```diff
@@ -0,0 +1,257 @@
+        parsed.height = (h == null) ? "" : heightRegex.exec(stringToBeParsed).groups.height;
+        parsed.gravity = (g == null) ? "" : gravityRegex.exec(stringToBeParsed).groups.gravity;
+        parsed.crop = (c == null) ? "" : cropRegex.exec(stringToBeParsed).groups.crop;
+
+        return parsed;
+    }
+
+    parseSourceString(e, srcString) {
+        let selected = this.regexParser(srcString);
+
+        document.getElementById('asset-selector').value = selected.image;
+        this.formData.codename = document.getElementById('asset-selector').value;
+        document.getElementById('minWidth').value = selected.minWidth;
+        this.formData.minWidth = document.getElementById('minWidth').value;
+        document.getElementById('maxWidth').value = selected.maxWidth;
+        this.formData.maxWidth = document.getElementById('maxWidth').value;
+        document.getElementById('width').value = selected.width;
+        this.formData.width = document.getElementById('width').value;
+        document.getElementById('height').value = selected.height;
+        this.formData.height = document.getElementById('height').value;
+        document.getElementById('gravity').value = selected.gravity;
+        this.formData.gravity = document.getElementById('gravity').value;
+        document.getElementById('crop').value = selected.crop;
+        this.formData.crop = document.getElementById('crop').value;
+
+        this.buildItemPreview();
+    }
+
+    removeItem(e, sourcetag) {
+        var element = document.getElementById(sourcetag);
+        element.parentNode.removeChild(element);
+
+        for (var i = 0; i < this.sources.length; i++) {
+            if (this.sources[i].sourceTag == sourcetag) {
+                this.sources.splice(i, 1);
+            }
+
+            if (sourcetag == this.formData.sourceTag) {
+                document.getElementById('maxWidth').value = "";
+                this.formData.maxWidth = "";
+                document.getElementById('minWidth').value = "";
+                this.formData.minWidth = "";
+                document.getElementById('width').value = "";
+                this.formData.width = "";
+                document.getElementById('height').value = "";
+                this.formData.height = "";
+                document.getElementById('gravity').value = "";
+                this.formData.gravity = "";
+                document.getElementById('crop').value = "";
+                this.formData.crop = "";
+            }
+        }
+
+        this.updateValue();
+        e.stopPropagation()
+    }
+
+    updateValues(e) {
+        for (name in this.formData) {
+            if (e.target.name == name) {
+                this.formData[name] = e.target.value;
+            }
+        }
+    }
+
+    initImagePreview() {
+        document.getElementById('image').src = `https://res.cloudinary.com/${this.options.acctName}/image/upload/${this.formData.codename}`;
```

Comment:
Use cloudinaryManager to get the image string.

Reply:
done